app/_plugins/single_source_generator.rb

```diff
@@ -0,0 +1,61 @@
+module SingleSource
+  class Generator < Jekyll::Generator
+    priority :highest
+    def generate(site)
+      data = SafeYAML.load(File.read("app/_data/docs_nav_deck_1.12.x.yml"))
+      return unless data['generate']
+      createPages(data['items'], site, data['product'], data['version'])
+    end
+
+    def createPages(data, site, product, version)
+      data.each do |v,k|
+        if (v['url'] && !v['absolute_url']) || v['src']
+          # Is it an in-page link? If so, skip it
+          next if v['url'] && v['url'].include?("/#")
+          site.pages << SingleSourcePage.new(site, v['src'], v['url'], product, version)
+        end
+        if v['items']
+          createPages(v['items'], site, product, version)
+        end
+      end
+    end
+  end
+
+  class SingleSourcePage < Jekyll::Page
+    def initialize(site, src, dest, product, version)
+      # If there's no 'src' key provided, assume it's the same as the url
+      src = dest unless src
+      if src[0] === "/"
+        src = src[1..-1]
+      end
+
+      # Remove trailing slashes
+      src.chomp!("/")
```

Comment:
This is building paths on disk, rather than in the URL. `src` is a path on disk, and I remove the trailing slash so that when I add one later I know that the input is normalised.

Reply:
On an unrelated note, https://www.zachleat.com/web/trailing-slash/ is excellent reading on trailing slashes

Reply:
That is excellent reading, and mainly what I get out of that is consistency is king. And we... are not consistent. It might be unrelated to this PR, but this consistency is something we'll have to work on in the future anyway, so thanks for the reading material 😄